tests/monitoring/monitoring.go

```diff
@@ -251,17 +254,39 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", func() {
 		checks.SkipIfPrometheusRuleIsNotEnabled(virtClient)
 	})
 
-	Context("Alerts runbooks", func() {
-		It("Should have available URLs", func() {
-			alerts, err := getAlerts(virtClient)
+	Context("Kubevirt alert rules", func() {
+		BeforeEach(func() {
+			monv1 = virtClient.PrometheusClient().MonitoringV1()
+			prometheusRule, err = monv1.PrometheusRules(flags.KubeVirtInstallNamespace).Get(context.Background(), components.KUBEVIRT_PROMETHEUS_RULE_NAME, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			for _, alert := range alerts {
-				Expect(alert.Annotations).ToNot(BeNil())
-				url, ok := alert.Annotations["runbook_url"]
-				Expect(ok).To(BeTrue())
-				resp, err := http.Head(string(url))
-				Expect(err).ToNot(HaveOccurred())
-				Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+		})
+
+		It("[test_id:8821]should have all the requried annotations", func() {
+			for _, group := range prometheusRule.Spec.Groups {
+				for _, rule := range group.Rules {
+					if rule.Alert != "" {
+						Expect(rule.Annotations).To(HaveKeyWithValue("summary", Not(BeEmpty())),
+							fmt.Sprintf("%s summary is missing or empty", rule.Alert))
+						Expect(rule.Annotations).To(HaveKeyWithValue("runbook_url", Not(BeEmpty())),
+							fmt.Sprintf("%s runbook_url is missing or empty", rule.Alert))
+						checkRunbookUrlAvailability(rule)
+					}
+				}
+			}
+		})
+
+		It("[test_id:8822]should have all the requried labels", func() {
+			for _, group := range prometheusRule.Spec.Groups {
+				for _, rule := range group.Rules {
+					if rule.Alert != "" {
+						Expect(rule.Labels).To(HaveKeyWithValue("severity", BeElementOf("info", "warning", "critical")),
```

Comment:
@assafad Can you please encapsulate the logic that checks the required labels into a separate private function? something like `checkRequiredLabels`

Reply:
Done